core/server/middleware/middleware.js

```diff
@@ -325,6 +338,128 @@ middleware = {
         next();
     },
 
+    checkIsProtected: function (req, res, next) {
+        api.settings.read({context: {internal: true}, key: 'isPrivate'}).then(function (response) {
+            var pass = response.settings[0];
+            if (_.isEmpty(pass.value) || pass.value === 'false') {
+                res.isPrivateBlog = false;
+                return next();
+            }
+            res.isPrivateBlog = true;
+            return session({
+                maxAge: utils.ONE_MONTH_MS,
+                keys: ['isPrivateBlog']
```

Comment:
I'm not sure if this is the best thing to put for cookie session keys....
@ErisDS any better ideas on what to put for keys?
looking at the cookie-session documentation, I'm not sure what this does :/